#11811: Sanitize SQL inode parameter

oarrietadotcms opened the pull request.

oarrietadotcms committed:
#11811: Sanitize SQL inode parameter

jdotcms commented on dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategoryFactoryImpl.java:
```diff
@@ -736,6 +736,8 @@ public void sortTopLevelCategories()  throws DotDataException {
 	}
 
 	public void sortChildren(String inode)  throws DotDataException {
+        inode = SQLUtil.sanitizeParameter( inode );
```
Just a question from my ignorance; why are we sanitizing an inode?

wezell replied:
Oscar, this is ok, but a better fix would be to make this method use a PreparedStatement instead of just injecting the inode value

https://github.com/dotCMS/core/blob/master/dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategorySQL.java#L39

oarrietadotcms replied:
That was my first thought, the thing is that we would need a mejor refactor on that logic, first we need to change [this method](https://github.com/dotCMS/core/blob/master/dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/MySQLCategorySQL.java#L22-L28), the signature for every other DBs, in order to return a PreparedStatement not a String, then we would need to change the other methods as well to avoid having 2 patterns in the same functionality. I couldn't find any integration test around that logic and at that point I changed my mind and did the implemented the safer option: SQL Sanitize Parameter. 

@dsilvam and @wezell let me know if you want to use the PreparedStatement anyway and I will develop it and create the tests around it.

oarrietadotcms replied:
@jdotcms the _inode_ is a servlet param, so it could be anything, not just the UUID.

dsilvam replied:
@oarrietadotcms I just walked through the changes needed and, even though they are a bit tedious, I vote for changing to PreparedStatements

oarrietadotcms committed:
#11811: Using PreparedStatement instead of Sanitize and adding integration test.
CategoryAPITest.java: Add testSortChildren() test.
CategoryFactoryImpl.java: Execute PreparedStatement and extract duplicated code.
*CategorySQL.java: Replacing the inode concat with ? char to be used in the Prepared Statement.

oarrietadotcms commented:
@dsilvam PR updated with latest feedback

oarrietadotcms committed:
#11811: Test cleaning should be under the finally
